### PR TITLE
Clarify the migration guide a bit

### PR DESCRIPTION
**Pull request by djbe**
Discussed here: https://github.com/SwiftGen/SwiftGen/pull/320#discussion_r131523142

### CODE REVIEW

File: Documentation/MigrationGuide.md [hunk 1]
```diff
@@ -5,52 +5,55 @@ The two general themes for this version are:
 - Templates now reside in the subfolder corresponding to their subcommand (`colors`, `fonts`, ...) intead of the filename being prefixed with it.
 - The `default` template doesn't exist anymore, templates now specify which swift version they support.
 
-Below is a list of renamed and removed templates, grouped by subcommand. If your template isn't listed, you don't need to do anything.
+Below is a list of renamed and removed templates, grouped by subcommand. If your template hasn't been renamed or removed, you don't need to do anything. You still might want to review the documentation for that template, as there might be new features you may be interested in.
```
Comment (AliSoftware) [inline Documentation/MigrationGuide.md]: Maybe use the occasion of that sentence to explain our emoji key?

```
Below is a list of renamed ("➡️") and removed ("❌") templates
[…]
you don't ned to do anything ("✅").
```

**Review by AliSoftware — COMMENTED** [review]
Didn't manually check that you listed/added all the templates, but LGTM.

**Comment by AliSoftware** [comment]
🤔💭 Maybe it would be less scary and make more sense to use ➡️ for templates that have been merged. After all, if a template has been deleted because it has been merged with another, like iOS/macOS variants for storyboards, that would make more sense to say that both have been renamed to the same name `swift3`, using ➡️ as in "if you used that old name before, then you just have to use this new name now"?

File: Documentation/MigrationGuide.md [hunk 2]
```diff
@@ -5,52 +5,55 @@ The two general themes for this version are:
...
 
 | Old | New | Reason |
 | --- | --- | ------ |
-| all-values | **deleted** | The other templates by default now generate an `allValues` static constant |
-| default | **deleted** | Deprecated by `dot-syntax` (now called `swift2`) |
-| dot-syntax | swift2 | |
-| dot-syntax-swift3 | swift3 | |
-| swift3 | **deleted** | Deprecated by `dot-syntax-swift3` (now called `swift3`) |
+| `all-values` | ❌ _deleted_ | The other templates by default now generate an `allValues` static constant |
+| `default` | ❌ _deleted_ | Deprecated by `dot-syntax` (now called `swift2`) |
```
Comment (AliSoftware) [inline Documentation/MigrationGuide.md]: ➡️ `swift2`

File: Documentation/MigrationGuide.md [hunk 3]
```diff
@@ -5,52 +5,55 @@ The two general themes for this version are:
...
+| `dot-syntax` | ➡️ `swift2` | |
+| `dot-syntax-swift3` | ➡️ `swift3` | |
+| `swift3` | ❌ _deleted_ | Deprecated by `dot-syntax-swift3` (now called `swift3`) |
```
Comment (AliSoftware) [inline Documentation/MigrationGuide.md]: Mmmh that's a tough case this one. In a sense, It has been "renamed to the same name" 😄
Maybe use a warning emoji to tell there's no name change required from the user but they have to be aware of a change in the output?

```
| `swift3` | ⚠️ swift3 | Now integrates the recursive features of the previously named `dot-syntax-swift3` template |
```

File: Documentation/MigrationGuide.md [hunk 4]
```diff
@@ -5,52 +5,55 @@ The two general themes for this version are:
...
 | --- | --- | ------ |
-| default | swift2 | |
-| lowercase | **deleted** | No longer needed since we prefix classes with their module |
-| osx-default | **deleted** | Unified with the iOS template, just use `swift2` |
-| osx-lowercase | **deleted** | No longer needed since we prefix classes with their module |
-| osx-swift3 | **deleted** | Unified with the iOS template, just use `swift3` |
-| uppercase | **deleted** | No longer needed since we prefix classes with their module |
+| `default` | ➡️ `swift2` | |
+| `lowercase` | ❌ _deleted_ | No longer needed since we prefix classes with their module |
+| `osx-default` | ❌ _deleted_ | Unified with the iOS template, just use `swift2` |
```
Comment (AliSoftware) [inline Documentation/MigrationGuide.md]: ➡️ `swift2`

File: Documentation/MigrationGuide.md [hunk 5]
```diff
@@ -5,52 +5,55 @@ The two general themes for this version are:
...
+| `osx-lowercase` | ❌ _deleted_ | No longer needed since we prefix classes with their module |
+| `osx-swift3` | ❌ _deleted_ | Unified with the iOS template, just use `swift3` |
```
Comment (AliSoftware) [inline Documentation/MigrationGuide.md]: ➡️ `swift3`

**Review by AliSoftware — COMMENTED** [review]
Some proposals for "deleted because deprecated in favour of another one" cases, lmkwyt

File: Documentation/MigrationGuide.md [hunk 6]
```diff
@@ -5,52 +5,55 @@ The two general themes for this version are:
...
 
 | Old | New | Reason |
 | --- | --- | ------ |
-| all-values | **deleted** | The other templates by default now generate an `allValues` static constant |
-| default | **deleted** | Deprecated by `dot-syntax` (now called `swift2`) |
-| dot-syntax | swift2 | |
-| dot-syntax-swift3 | swift3 | |
-| swift3 | **deleted** | Deprecated by `dot-syntax-swift3` (now called `swift3`) |
+| `allvalues` | ➡️ `swift2`/`swift3` | All templates by default now generate an `allValues` static constant |
+| `default` | ❌ _deleted_ | Deprecated by `dot-syntax` (now called `swift2`) |
```
Comment (AliSoftware) [inline Documentation/MigrationGuide.md]: ➡️ `swift2` then?

Comment (djbe) [reply Documentation/MigrationGuide.md]: Same warning as the swift3 one then?

Comment (AliSoftware) [reply Documentation/MigrationGuide.md]: Ah yeah maybe indeed.

Comment (djbe) [reply Documentation/MigrationGuide.md]: We have to be careful though. This is a breaking change for the users, as they'll have to change their code to match the changes (from non-dot-syntax to dot-syntax).

Comment (AliSoftware) [reply Documentation/MigrationGuide.md]: Or use "➡️⚠️" ? ^^

Comment (AliSoftware) [reply Documentation/MigrationGuide.md]: Good point (even though… who uses swift2 anymore? 😜)

Comment (djbe) [reply Documentation/MigrationGuide.md]: Now that's becoming some symbol bonanza 😄 Let's leave it at a warning, it's a breaking change.

**Pull request merged**